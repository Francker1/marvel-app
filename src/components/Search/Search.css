.search-container {
  position: relative;
  margin: 20px 0 12px 0;
}

.search-input {
  border: none;
  border-bottom: 1px solid var(--blackColor);
  border-radius: 0;
  box-shadow: none;
  color: var(--blackColor);
  font-family: var(--primaryFont);
  padding: 12px 20px 8px 40px;
  width: 100%;
}

.search-input::placeholder {
  color: var(--greyColor);
}

.search-input,
.search-input::placeholder {
  text-transform: uppercase;
  font-size: 18px;
}

.search-input:focus {
  border-color: var(--blackColor);
  outline: none;
}

.search-icon {
  position: absolute;
  left: 10px;
  top: 50%;
  transform: translateY(-50%);
  width: 20px;
  height: 20px;
}