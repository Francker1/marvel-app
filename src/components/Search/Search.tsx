import React from 'react';
import SearchIcon from '../../assets/img/Search-icon.svg';
import './Search.css';

interface SearchProps {
  searchTerm: string;
  handleSearchChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

const Search: React.FC<SearchProps> = ({ searchTerm, handleSearchChange }) => {
  return (
    <div className="search-container">
      <img src={SearchIcon} alt="Search Icon" className="search-icon" />
      <input
        type="text"
        placeholder="Search a character"
        className="search-input"
        value={searchTerm}
        onChange={handleSearchChange}
      />
    </div>
  );
}

export default Search;
