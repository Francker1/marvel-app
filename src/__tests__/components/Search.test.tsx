import { render, screen, fireEvent } from '@testing-library/react';
import Search from '../../components/Search/Search';
import { expect, test, vi } from 'vitest';

test('renders correctly with the given search term', () => {
  const { asFragment } = render(<Search searchTerm="Spi" handleSearchChange={() => {}} />);
  const input = screen.getByPlaceholderText('Search a character');
  expect(input).toBeInTheDocument();
  expect(input).toHaveValue('Spi');
  expect(asFragment()).toMatchSnapshot(); // Crear snapshot
});

test('calls handleSearchChange on input change', () => {
  const handleSearchChange = vi.fn();
  render(<Search searchTerm="" handleSearchChange={handleSearchChange} />);
  const input = screen.getByPlaceholderText('Search a character');
  fireEvent.change(input, { target: { value: 'Iron Man' } });
  expect(handleSearchChange).toHaveBeenCalledWith(expect.any(Object));
});
