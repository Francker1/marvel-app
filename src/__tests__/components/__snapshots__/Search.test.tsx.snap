// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders character card correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-braxZu hxZAuk"
  >
    <a
      class="sc-gJhJTp hgPeRr"
      href="/character/1"
    >
      <img
        alt="Spider-Man"
        class="sc-elDIKY cMlytH"
        src="http://example.com/spider-man.jpg"
      />
    </a>
    <div
      class="sc-fQpRED hEHhrj"
    >
      <span
        class="sc-dsLQwm lnMvxE"
      >
        Spider-Man
      </span>
      <span
        class="sc-iKTcqh dSEPIF"
      >
        <img
          alt="Add to Favorites"
          src="/src/assets/img/Heart-icon-empty.svg"
        />
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders correctly with the given search term 1`] = `
<DocumentFragment>
  <div
    class="search-container"
  >
    <img
      alt="Search Icon"
      class="search-icon"
      src="/src/assets/img/Search-icon.svg"
    />
    <input
      class="search-input"
      placeholder="Search a character"
      type="text"
      value="Spi"
    />
  </div>
</DocumentFragment>
`;
